huma.go

```diff
@@ -1226,7 +1226,9 @@ func Register[I, O any](api API, op Operation, handler func(context.Context, *I)
 						// second time is faster than `mapstructure.Decode` or any of the other
 						// common reflection-based approaches when using real-world medium-sized
 						// JSON payloads with lots of strings.
-						f := v.Field(inputBodyIndex)
+						// Body struct field is retrieved by name to work with embedded structs,
+						// as its existence has already been verified.
+						f := v.FieldByName("Body")
```

Comment:
Nice find for this bug! One small nit: based on benchmarks it is much faster to get a field by index, so I think if the body is in an embedded struct we should store a slice of indexes to access it rather than using `FieldByName`. Otherwise I think this PR is great!

Reply:
That makes sense, I patched the PR following your suggestion. I'm confused about why the code coverage check fails tho